### PR TITLE
RFR -add review spec and model

### PR DESCRIPTION
**Pull request by sammcgrail**

### CODE REVIEW

File: spec/models/review_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+describe Review do
+  it { should belong_to :rubygem }
+
+  it { should have_valid(:rating).when(5) }
+  it { should have_valid(:rating).when(4) }
+  it { should have_valid(:rating).when(3) }
+  it { should have_valid(:rating).when(2) }
+  it { should have_valid(:rating).when(1) }
```
Comment (davidrf) [inline spec/models/review_spec.rb]: refactor lines 6-10
`it { should have_valid(:rating).when(1, 2, 3, 4, 5) }`

File: spec/models/review_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  it { should_not have_valid(:rating).when(nil) }
+  it { should_not have_valid(:rating).when("string") }
+  it { should_not have_valid(:rating).when(-1) }
```
Comment (davidrf) [inline spec/models/review_spec.rb]: Same comment as above

File: spec/models/rubygem_spec.rb [hunk 3]
```diff
@@ -8,19 +8,10 @@
...
   it { should have_valid(:review_count).when(34) }
-  it { should_not have_valid(:review_count).when(nil) }
-  it { should_not have_valid(:review_count).when("two") }
 
   it { should have_valid(:user_id).when(2) }
-  it { should_not have_valid(:user_id).when("five") }
-  it { should_not have_valid(:user_id).when(nil) }
 
   it { should have_valid(:link_to_rubygems_url).when("http://blahblahblah.com") }
   it { should have_valid(:link_to_rubygems_url).when("https://blahblahblah.com") }
```
Comment (davidrf) [inline spec/models/rubygem_spec.rb]: Why is this here twice??

File: spec/models/rubygem_spec.rb [hunk 4]
```diff
@@ -8,19 +8,10 @@
...
-  it { should_not have_valid(:average_rating).when(10) }
-  it { should_not have_valid(:average_rating).when(0) }
-  it { should_not have_valid(:average_rating).when(nil) }
-  it { should_not have_valid(:average_rating).when(-2) }
 
   it { should have_valid(:review_count).when(34) }
-  it { should_not have_valid(:review_count).when(nil) }
-  it { should_not have_valid(:review_count).when("two") }
 
   it { should have_valid(:user_id).when(2) }
```
Comment (davidrf) [inline spec/models/rubygem_spec.rb]: Use `:user` instead of `:user_id`

`it { should have_valid(:user).when(User.new) }

File: spec/models/rubygem_spec.rb (hunk 3 above)
Comment (sammcgrail) [inline spec/models/rubygem_spec.rb]: for https. I guess this is unneeded, thanks

File: app/models/rubygem.rb [hunk 5]
```diff
@@ -1,11 +1,8 @@
 class Rubygem < ActiveRecord::Base
   has_many :reviews
   belongs_to :user
-
```
Comment (davidrf) [inline app/models/rubygem.rb]: Keep a space between the associations and validations.
